src/main/java/com/aws/greengrass/deployment/bootstrap/BootstrapManager.java

```
@@ -503,27 +521,33 @@ public void deleteBootstrapTaskList(Path persistedTaskFilePath) throws IOExcepti
     /**
      * Persist the bootstrap task list from file.
      *
-     * @param persistedTaskFilePath path to the persisted file of bootstrap tasks
-     * @throws IOException on I/O error
-     * @throws ClassNotFoundException deserialization of the file content fails
+     * @param persistedTaskFilePath
+     *            path to the persisted file of bootstrap tasks
+     * @throws IOException
+     *             on I/O error
+     * @throws ClassNotFoundException
+     *             deserialization of the file content fails
      */
     @SuppressWarnings("PMD.PreserveStackTrace")
     public void loadBootstrapTaskList(Path persistedTaskFilePath) throws IOException {
         Objects.requireNonNull(persistedTaskFilePath);
 
         CommitableReader.of(persistedTaskFilePath).read(in -> {
             bootstrapTaskStatusList.clear();
-            bootstrapTaskStatusList.addAll(SerializerFactory.getFailSafeJsonObjectMapper()
-                    .readValue(in, new TypeReference<List<BootstrapTaskStatus>>(){}));
+            bootstrapTaskStatusList.addAll(SerializerFactory.getFailSafeJsonObjectMapper().readValue(in,
```

Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The `ObjectMapper.readValue()` method can throw a `JsonProcessingException`, but this exception is not handled properly here. By simply logging the error and returning an empty value, the exception is ignored. This can mask potential JSON parsing issues, leading to unexpected application behavior or data corruption because the system may continue with invalid or incomplete data. To fix this, catch `JsonProcessingException` and either handle it appropriately or rethrow it as a custom runtime exception. [https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)